core/src/main/java/com/ibm/wala/ipa/cha/ClassHierarchy.java

```diff
@@ -706,6 +707,86 @@ private void recursiveStringify(Node n, StringBuilder buffer) {
     }
   }
 
+  /*
+   * Puts ClassHierarchy into a JSON readable variable that maps the class to a list of its
+   * subclasses
+   */
+  public String toJson() {
+    // initialize variables (subclass and dag) to store important values like hashmap of each
+    // iteration and a list of all maps of class to subclasses
+    HashMap<String, Set<String>> subclass = new HashMap<>();
+    Iterator<Node> children = root.getChildren();
+    // Goes through all subclasses of the root class, and adds the subclass hashmap to the
+    // list of all subclass hashmaps
+    while (children.hasNext()) {
+      helperToJson(children.next(), subclass);
+    }
+    // Removes unnecesarry parts from name of the class
+    String key = nodeToString(root);
+    // inserting the root class to its subclasses into the main hashmap while removing
+    // unnecessary substrings from the class name
+    Iterator<Node> rootChildren = root.getChildren();
+    Set<String> finalRootChildren = new HashSet<>();
+    while (rootChildren.hasNext()) {
+      finalRootChildren.add(nodeToString(rootChildren.next()));
+    }
+    subclass.put(key, finalRootChildren);
+    Gson gson = new Gson();
+    return gson.toJson(subclass);
+  }
+
+  /*
+   * helper function to toJson that performs recursion to go through all of the DAG
+   */
+  private void helperToJson(Node n, HashMap<String, Set<String>> hash) {
```

Comment:
I think the code in this method is a bit buggy.  I think the guarantee we want is that when `helperToJSON` finishes, the entire subtree of the classhierarchy rooted at `n` is in the `classNameToSubclassName` map.  So, the steps should be:
1. Compute `key = nodeToString(n)`.  If `key` is already in the map, return.
2. Create a set (or list) `subclassNames`.  Iterate through the children, and for each `child`, recursively call `helperToJSON(child)` and also add `nodeToString(child)` to `subclassNames`. 
3. After the loop, do `classToSubclassNames.put(key,subclassNames)`.

If you do things this way, you can get rid of special-case code for the root node in `toJSON` and just call `helperToJson` on the root, and everything should work.